samples/ComplexParametric.java

```diff
@@ -43,7 +43,7 @@ interface SuperSuper<T extends @Nullable Object> {
 
     void checkNeverNull(Lib<? extends Object> lib);
 
-    <U> void checkUnspecNull(Lib<@NullnessUnspecified U> lib);
+    <U extends @Nullable Object> void checkUnspecNull(Lib<@NullnessUnspecified U> lib);
```

Comment:
Without the more flexible bound, there are warnings about invalid inferred method type arguments.
As the bound of `U` doesn't really matter, I think this doesn't change anything substantially.
Let me know if this isn't correct.

Reply:
I think that https://github.com/jspecify/jspecify/blob/8989d39851ae59b92db141b051ebdea37fbe4845/samples/ComplexParametric.java#L21-L25 is enough reason to worry too much about this file in general :)